### ISSUE TITLE
CommonJS Types appear incorrect

### ISSUE THREAD

**Issue by charliematters** [tailwindlabs/headlessui#2838]
**What package within Headless UI are you using?**

@headlessui/react

**What version of that package are you using?**

1.7.17

**What browser are you using?**

Typescript

**Reproduction URL**
https://github.com/charliematters/tshy-repo

**Describe your issue**

I've been trying to use headless ui as part of an internal component package, which is compiled to ESM and CJS versions by `tshy`.

When trying to produce the CJS bundle I get the same error as this closed issue: https://github.com/tailwindlabs/headlessui/issues/2386#issuecomment-1663921719

I [asked the author](https://x.com/izs/status/1724131070264905961?s=20) of the`tshy` package for some advice and he pointed me towards this page, which reports that the CJS types are just copies of the ESM ones, which apparently will not work:

https://arethetypeswrong.github.io/?p=%40headlessui%2Freact%401.7.17

**Comment by thecrypticace** [comment]
Hey, comically enough our type files were technically correct for CJS types it's just the resolution that was wrong. I did a quick and dirty fix to get things working.

Also found an ESM typing issue in our `@headlessui/tailwindcss` package because of this work too which I've fixed.

This will be available in the next tagged release. It'll also available in our insiders build in a few minutes:
```
npm install @headlessui/react@insiders
npm install @headlessui/vue@insiders
```

Thanks!
